### PR TITLE
Wrapper for perturbation data

### PR DESCRIPTION
**Pull request by davidsebfischer**
Perturbation data can create scenarios with large number of groups which may require pairwise comparisons. The pairwise comparison matrices may be too large to fully evaluate, it therefore makes sense to handle this in a separate wrapper that fits a model as in the pairwise() scenario but which performs the tests lezyly, ie which only returns the required comparisons. I added such a DET class for lazy testing for the z-test and a wrapper de.test.perturb().

### CODE REVIEW

**Comment by Hoeze** [comment]
Hm, wouldn't be a `lazy=True` option in `vs_rest` be sufficient for this?

**Comment by davidsebfischer** [comment]
I will add that to pairwise, not that this is not vs rest but a subset of pairwise comparisons. I guess a jupyter notebook with an example is better than an additional wrapper in this case.

File: diffxpy/testing/base.py [hunk 1]
```diff
@@ -1310,6 +1318,339 @@ def summary_pair(self, group1, group2,
...
+    def summary_pairs(self, groups0, groups1=None,
+                     qval_thres=None, fc_upper_thres=None,
+                     fc_lower_thres=None, mean_thres=None,
+                     **kwargs) -> pd.DataFrame:
+        """
+        Summarize differential expression results of a set of
+        pairwise comparisons into an output table.
+
+        :param groups0: First set of groups in pair-wise comparison.
+        :param groups0: Second set of groups in pair-wise comparison.
```
Comment (Hoeze) [inline diffxpy/testing/base.py]: 2x `groups0` => typo?

**Comment by Hoeze** [comment]
Looks OK for me :+1:

File: diffxpy/testing/base.py (hunk 1 above)
Comment (davidsebfischer) [reply diffxpy/testing/base.py]: Fixed this and other occurences in https://github.com/theislab/diffxpy/commit/17c77184523bf8238a93eff7e5117d27a2b85fd6.

**Pull request merged**